### DISCUSSION TITLE
p:selectOneRadio change by another component

### DISCUSSION THREAD

**Discussion by Alirzrezaei** [primefaces/community#1406] [PrimeFaces] [answered]
I have a scenario in which I am using a p:selectOneRadio. The selectItems are set in the Bean class with values and labels. 

![radio btns](https://github.com/primefaces/community/assets/32749480/300c7932-a7b9-4ff6-8e50-27ab00887020)

My data are in dataTable and there is a column for selectionMode="multiple". The Deselect All radio button will uncheck all of the checkboxes in that column with the following code. 

 ```
var dataTableWidget = PF('item');
dataTableWidget.unselectAllRows();
```

when the user selects again a row, the selected radio button should be selected. I have written some codes and see in the inspection area that the attribute is changed to true but on the screen the Deselect All button is selected. 

```
let radioBtn = document.getElementById("data:agentSelectMethod");
var options = radioBtn.getElementsByTagName('input');
options[1].checked = true;
options[1].ariaChecked = true;
options[1].ariaSelected = true;
```

Note: a div with role="radio" is generated automatically in frontend with the aria-checked="true/false" and also changed the attribute for this. but still, the radio button is the third not the second.
        
 ```
   var first = radioBtn.childNodes[0];
	var second = radioBtn.childNodes[1];
	var third = radioBtn.childNodes[2];
	first.ariaChecked= false;
	second.ariaChecked = true;
third.ariaChecked= false;
```

**Comment by melloware** [comment]
Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Reply by Alirzrezaei** [reply, marked as answer]
The problem is solved easily. I thought hard but the solution was easy.

solution was to set the item value in the backend a update the component. 

setSelectItemValue("item2");
PrimeFaces.current().ajax().update("oneSelectRadio_ID");
